src/MACHDYN/atom_vec_smd.cpp

```diff
@@ -201,7 +201,7 @@ void AtomVecSMD::write_data_restricted_to_general()
 
   int nlocal = atom->nlocal;
   memory->create(x0_hold,nlocal,3,"atomvec:x0_hold");
-  if (nlocal) memcpy(&x0_hold[0][0],&x0[0][0],3*nlocal*sizeof(double));
+  if (nlocal) memcpy(&x0_hold[0][0],&x0[0][0],(3*nlocal*sizeof(double))&memcpymask);
```

Comment:
@sjplimp @stanmoore1 this set of changes (inspired by a similar change in libfmt to work around the same kind of compiler issue) should silence the bogus compiler warnings. Please let me know if I missed a spot.